src/rpc/methods/eth.rs

```diff
@@ -1373,6 +1283,73 @@ impl RpcMethod<0> for EthSyncing {
     }
 }
 
+pub enum EthGetCode {}
+impl RpcMethod<2> for EthGetCode {
+    const NAME: &'static str = "Filecoin.EthGetCode";
+    const PARAM_NAMES: [&'static str; 2] = ["eth_address", "block_number_or_hash"];
+    const API_VERSION: ApiVersion = ApiVersion::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (EthAddress, BlockNumberOrHash);
+    type Ok = Bytes;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (eth_address, block_number_or_hash): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let ts = tipset_by_block_number_or_hash(&ctx.chain_store, block_number_or_hash)?;
+        let to_address = FilecoinAddress::try_from(&eth_address)?;
+        let actor = ctx
+            .state_manager
+            .get_required_actor(&to_address, *ts.parent_state())?;
+        // Not a contract. We could try to distinguish between accounts and "native" contracts here,
+        // but it's not worth it.
+        if !fil_actor_interface::is_evm_actor(&actor.code) {
+            Ok(Default::default())
+        } else {
+            let message = Message {
+                from: FilecoinAddress::SYSTEM_ACTOR,
+                to: to_address,
+                method_num: METHOD_GET_BYTE_CODE,
+                gas_limit: BLOCK_GAS_LIMIT,
+                ..Default::default()
+            };
+            let mut api_invoc_result = None;
+            for ts in ts.chain_arc(ctx.store()) {
+                match ctx.state_manager.call(&message, Some(ts)) {
+                    Ok(res) => {
+                        api_invoc_result = Some(res);
+                        break;
+                    }
+                    Err(e) => tracing::warn!(%e),
+                }
+            }
+            if let Some(api_invoc_result) = api_invoc_result {
+                if let Some(msg_rct) = api_invoc_result.msg_rct {
+                    if api_invoc_result.error.is_empty() {
+                        let get_bytecode_return: GetBytecodeReturn =
+                            fvm_ipld_encoding::from_slice(msg_rct.return_data().as_slice())?;
+                        if let Some(cid) = get_bytecode_return.0 {
+                            Ok(Bytes(ctx.store().get_required(&cid)?))
+                        } else {
+                            Ok(Default::default())
+                        }
+                    } else {
+                        Err(
+                            anyhow::anyhow!("GetBytecode failed: {}", api_invoc_result.error)
+                                .into(),
+                        )
+                    }
+                } else {
+                    Err(anyhow::anyhow!("no message receipt").into())
+                }
+            } else {
+                Err(anyhow::anyhow!("failed to call GetBytecode").into())
+            }
```

Comment:
Let-else may or may not be nicer here:
```suggestion
            let Some(api_invoc_result) = api_invoc_result else {
                return Err(anyhow::anyhow!("no message receipt").into());
            };
            let Some(msg_rct) = api_invoc_result.msg_rct else {
                return Err(anyhow::anyhow!("no message receipt").into());
            };
            if !api_invoc_result.error.is_empty() {
                return Err(
                    anyhow::anyhow!("GetBytecode failed: {}", api_invoc_result.error).into(),
                );
            }

            let get_bytecode_return: GetBytecodeReturn =
                fvm_ipld_encoding::from_slice(msg_rct.return_data().as_slice())?;
            if let Some(cid) = get_bytecode_return.0 {
                Ok(Bytes(ctx.store().get_required(&cid)?))
            } else {
                Ok(Default::default())
            }
```